### PR TITLE
Planning: separate  CUDA and non CUDA distance approach interface in open space planner

### PR DESCRIPTION
**Pull request by Capri2014**

### CODE REVIEW

File: modules/planning/open_space/trajectory_smoother/BUILD [hunk 1]
```diff
@@ -98,6 +98,7 @@ cc_library(
     ],
     deps = [
         "distance_approach_ipopt_interface",
+        "distance_approach_ipopt_cuda_interface",
```
Comment (xiaoxq) [inline modules/planning/open_space/trajectory_smoother/BUILD]: Should add leading ":"

File: modules/planning/open_space/trajectory_smoother/distance_approach_interface.h [hunk 2]
```diff
@@ -0,0 +1,127 @@
+/******************************************************************************
+ * Copyright 2017 The Apollo Authors. All Rights Reserved.
```
Comment (xiaoxq) [inline modules/planning/open_space/trajectory_smoother/distance_approach_interface.h]: 2019, as well as other new files.

Comment (Capri2014) [reply modules/planning/open_space/trajectory_smoother/distance_approach_interface.h]: done

File: modules/planning/open_space/trajectory_smoother/BUILD (hunk 1 above)
Comment (Capri2014) [reply modules/planning/open_space/trajectory_smoother/BUILD]: done

**Pull request merged**